wishlists.py

```diff
@@ -170,14 +170,29 @@ def clear_wishlist(wishlist_id):
                 return jsonify(message), HTTP_404_NOT_FOUND
         return db.retrieve_wishlist(wishlist_id), HTTP_200_OK
     except WishlistException:
+
         message = { 'error' : 'Wishlist %s was not found' % wishlist_id }
         return jsonify(message), HTTP_404_NOT_FOUND
 
 
+@app.route('/wishlists/search', methods=['GET'])
+def search_wishlists():
+    """
+    The route for searching items with specific keyword or ID.
+    """
+    try:
+        query = request.args.get('q', None)
+        uid = request.args.get('user_id',None)
+        print query
+        print uid
+        items = db.search_all_items(query,uid)
+        return items, HTTP_200_OK
+    except ItemException:
+        return jsonify(message='Item with query %s not found'%query), HTTP_404_NOT_FOUND
 if __name__ == '__main__':
 
     # Pull options from environment
     debug = os.getenv('DEBUG', 'False') == 'True'
     port = os.getenv('PORT', '5000')
-	
-    app.run(host='0.0.0.0', port=int(port), debug=debug)
+
+    app.run(host='0.0.0.0', port=int(port), debug=True)
```

Comment:
The application should *not* run in debug mode by default.  If you changed this value to test things, then you should change it back.  If you want to force debug mode, you can always `export DEBUG=True` before running the program.